packages/scheduler/src/forks/SchedulerHostConfig.post-task.js

```diff
@@ -5,246 +5,117 @@
+      if (!hasMoreWork) {
+        isMessageLoopRunning = false;
+        scheduledHostCallback = null;
       } else {
-        // There's still time left in the frame.
-        return false;
+        // If there's more work, schedule the next message event at the end
+        // of the preceding one.
+        postTask(performWorkUntilDeadline);
       }
-    };
-
-    requestPaint = function() {
-      needsPaint = true;
-    };
+    } catch (error) {
+      // If a scheduler task throws, exit the current browser task so the
+      // error can be observed.
+      postTask(performWorkUntilDeadline);
+      throw error;
+    }
   } else {
-    // `isInputPending` is not available. Since we have no way of knowing if
-    // there's pending input, always yield at the end of the frame.
-    shouldYieldToHost = function() {
-      return getCurrentTime() >= deadline;
-    };
-
-    // Since we yield every frame regardless, `requestPaint` has no effect.
-    requestPaint = function() {};
+    isMessageLoopRunning = false;
   }
+};
 
-  forceFrameRate = function(fps) {
-    if (fps < 0 || fps > 125) {
-      // Using console['error'] to evade Babel and ESLint
-      console['error'](
-        'forceFrameRate takes a positive int between 0 and 125, ' +
-          'forcing frame rates higher than 125 fps is not unsupported',
-      );
-      return;
-    }
-    if (fps > 0) {
-      yieldInterval = Math.floor(1000 / fps);
-    } else {
-      // reset the framerate
-      yieldInterval = 5;
-    }
-  };
-
-  const performWorkUntilDeadline = () => {
-    if (scheduledHostCallback !== null) {
-      const currentTime = getCurrentTime();
-      // Yield after `yieldInterval` ms, regardless of where we are in the vsync
-      // cycle. This means there's always time remaining at the beginning of
-      // the message event.
-      deadline = currentTime + yieldInterval;
-      const hasTimeRemaining = true;
-      try {
-        const hasMoreWork = scheduledHostCallback(
-          hasTimeRemaining,
-          currentTime,
-        );
-        if (!hasMoreWork) {
-          isMessageLoopRunning = false;
-          scheduledHostCallback = null;
-        } else {
-          // If there's more work, schedule the next message event at the end
-          // of the preceding one.
-          port.postMessage(null);
-        }
-      } catch (error) {
-        // If a scheduler task throws, exit the current browser task so the
-        // error can be observed.
-        port.postMessage(null);
-        throw error;
-      }
-    } else {
-      isMessageLoopRunning = false;
-    }
-    // Yielding to the browser will give it a chance to paint, so we can
-    // reset this.
-    needsPaint = false;
-  };
-
-  const channel = new MessageChannel();
-  const port = channel.port2;
-  channel.port1.onmessage = performWorkUntilDeadline;
+export const requestHostCallback = function(callback) {
```

Comment:
Nit: Switch to function declarations, i.e. `export function requestHostCallback`

I don't think it makes much of a difference and Closure and Rollup will optimize either form just as easily.

Mostly I just prefer to follow the convention we use in (most of) the other modules.